File: app/src/main/java/io/github/ovso/blackbox/di/FragmentBuilder.java
package io.github.ovso.blackbox.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import io.github.ovso.blackbox.ui.main.fragment.VideoFragment;
import io.github.ovso.blackbox.ui.main.fragment.di.VideoFragmentModule;
import io.github.ovso.blackbox.ui.main.fragment.di.VideoFragmentViewModule;
import javax.inject.Singleton;

@Module(includes = { AndroidSupportInjectionModule.class }) public abstract class FragmentBuilder {
  @Singleton @ContributesAndroidInjector(modules = {
      VideoFragmentModule.class,
      VideoFragmentViewModule.class
  })
  abstract VideoFragment contributeRepoFragment();
}

File: app/src/main/java/io/github/ovso/blackbox/di/FragmentBuilder.kt
package io.github.ovso.blackbox.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import io.github.ovso.blackbox.ui.main.fragment.VideoFragment
import io.github.ovso.blackbox.ui.main.fragment.di.VideoFragmentModule
import io.github.ovso.blackbox.ui.main.fragment.di.VideoFragmentViewModule
import javax.inject.Singleton

@Module(includes = arrayOf(AndroidSupportInjectionModule::class))
abstract class FragmentBuilder {
  @Singleton
  @ContributesAndroidInjector(
      modules = arrayOf(VideoFragmentModule::class, VideoFragmentViewModule::class)
  )
  internal abstract fun contributeRepoFragment(): VideoFragment
}